specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -6888,6 +7342,305 @@
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the disk name."
+        },
+        "caching": {
+          "$ref": "#/definitions/Caching",
+          "description": "Gets or sets the caching type."
+        },
+        "diskSizeGB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets the disk size in GB."
+        },
+        "managedDisk": {
+          "$ref": "#/definitions/ManagedDiskParameters",
+          "description": "Gets the managed disk details"
+        },
+        "diskRestorePoint": {
+          "$ref": "#/definitions/ApiEntityReference",
+          "description": "Gets or sets the disk restore point Id."
+        }
+      },
+      "description": "Describes an Operating System disk."
+    },
+    "RestorePointSourceVMDataDisk": {
+      "properties": {
+        "lun": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the logical unit number."
+        },
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the disk name."
+        },
+        "caching": {
+          "$ref": "#/definitions/Caching",
+          "description": "Gets or sets the caching type."
+        },
+        "diskSizeGB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the initial disk size in GB for blank data disks, and the new desired size for existing OS and Data disks."
+        },
+        "managedDisk": {
+          "$ref": "#/definitions/ManagedDiskParameters",
+          "description": "Gets the managed disk details"
+        },
+        "diskRestorePoint": {
+          "$ref": "#/definitions/ApiEntityReference",
+          "description": "Gets or sets the disk restore point Id."
+        }
+      },
+      "description": "Describes a data disk."
+    },
+    "RestorePointProvisioningDetails": {
+      "properties": {
+        "creationTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets the creation time of the restore point."
+        },
+        "totalUsedSizeInBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Gets the total size of the data in all the disks which are part of the restore point."
+        },
+        "statusCode": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets the status of the Create restore point operation."
+        },
+        "statusMessage": {
+          "type": "string",
+          "description": "Gets the status message of the Create restore point operation."
+        }
+      },
+      "description": "Restore Point Provisioning details."
+    },
+    "RestorePoint": {
+      "properties": {
+        "optionalProperties": {
+          "type": "object",
+          "description": "Optional properties to be passed for restore point creation (raw JSON)"
+        },
```

Comment:
We should not document this property as it is primarily used for extensibility for testing of new capabilities or features in the VM snapshot extension, external customers should not being using this property.

Reply:
i did not know that. will remove, thanks